src/System.Diagnostics.Process/src/System/Diagnostics/ProcessManager.OSX.cs

```diff
@@ -103,8 +105,76 @@ private static ProcessInfo CreateProcessInfo(int pid)
         /// <returns>The array of modules.</returns>
         internal static ModuleInfo[] GetModuleInfos(int processId)
         {
-            // We currently don't provide support for modules on OS X.
-            return Array.Empty<ModuleInfo>();
+            var modules = new List<ModuleInfo>();
+
+            // On Linux we can read from procfs.  On OS X we can get similar data by getting the output from vmmap.
+            // This is the same basic approach taken by the PAL in libcoreclr.
+            IntPtr popenStdout = Interop.Sys.POpen($"/usr/bin/vmmap -interleaved {processId} -wide", "r");
+            Debug.Assert(popenStdout != IntPtr.Zero, $"popen failed: {Interop.Sys.GetLastErrorInfo()}");
+            if (popenStdout != IntPtr.Zero) // if this failed, we'll just return an empty set of modules
+            {
+                try
+                {
+                    string line;
```

Comment:
Your comment specifies that line should be free'd by the caller but I don't see any Free calls...does line need to be free'd?